src/rio.h

```diff
@@ -107,6 +109,17 @@ typedef struct _rio rio;
 
 static inline size_t rioWrite(rio *r, const void *buf, size_t len) {
     if (r->flags & RIO_FLAG_WRITE_ERROR) return 0;
+    if (r->pre_flush_buffer) {
+        /* We have data that must be flushed before saving the current data.
+         * Lets flush it. */
+        sds pre_flush_buffer = r->pre_flush_buffer;
+        r->pre_flush_buffer = NULL;
+        size_t res = rioWrite(r, pre_flush_buffer, sdslen(pre_flush_buffer));
+        sdsfree(pre_flush_buffer);
+        if (!res) {
+             return 0;
+        }
+    }
```

Comment:
i'd rather avoid any changes to rio.h/c for this.
please look into implementing it when saving RDB_MODULE_OPCODE_STRING etc. i.e. in RM_SaveString and friends.